### ISSUE TITLE
sudden increase of  evaluation error rate in LSTMP training

### ISSUE THREAD

**Issue by Turan111** [Microsoft/CNTK#413]
Hi,

In LSTMP training on speech corpus about 140 hours I got sudden explosion of  evaluation error rate. 

```
Epoch[ 3 of 50]-Minibatch[56321-56330, 0.000000000019543%]: SamplesSeen = 2560; TrainLossPerSample =  1.44218750; EvalErr[0]PerSample = 0.39296875; TotalTime = 1.3378s; SamplesPerSecond = 1913.6
 Epoch[ 3 of 50]-Minibatch[56331-56340, 0.000000000019547%]: SamplesSeen = 2560; TrainLossPerSample =  1.56796875; EvalErr[0]PerSample = 0.40312500; TotalTime = 1.3372s; SamplesPerSecond = 1914.4
 Epoch[ 3 of 50]-Minibatch[56341-56350, 0.000000000019550%]: SamplesSeen = 2560; TrainLossPerSample =  1.31406250; EvalErr[0]PerSample = 0.35039063; TotalTime = 1.3350s; SamplesPerSecond = 1917.5
 Epoch[ 3 of 50]-Minibatch[56351-56360, 0.000000000019554%]: SamplesSeen = 2560; TrainLossPerSample = 12.34609375; EvalErr[0]PerSample = 0.72421875; TotalTime = 1.3360s; SamplesPerSecond = 1916.2
 Epoch[ 3 of 50]-Minibatch[56361-56370, 0.000000000019557%]: SamplesSeen = 2560; TrainLossPerSample =  9.83125000; EvalErr[0]PerSample = 0.94531250; TotalTime = 1.3435s; SamplesPerSecond = 1905.4
 Epoch[ 3 of 50]-Minibatch[56371-56380, 0.000000000019561%]: SamplesSeen = 2560; TrainLossPerSample =  8.65937500; EvalErr[0]PerSample = 0.92539063; TotalTime = 1.3380s; SamplesPerSecond = 1913.3
```

Until this moment error rate gradually reduced to 0.35, but after this moment it changed between the values 0.70 and 0.95.

And when I change readMethod from blockRandomize to rollingWindow and start training again from epoch 0, evaluation error rate didn't fall below 0.70.

**Comment by yzhang87** [comment]
That is gradient explosion. You need reduce the learning rate or set the gradient clipping threshold (e.g. 0.1).
